WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -72,6 +91,59 @@ public Pocket[] AutoSelectPockets()
 		return _allPockets.ToArray();
 	}
 
+	private bool TryGetBestKnownByRecipientPockets(Pocket[] knownByRecipientPockets, Money targetAmount, SmartLabel recipient, [NotNullWhen(true)] out Pocket[]? pockets)
+	{
+		pockets = null;
+
+		if (knownByRecipientPockets.Sum(x => x.Amount) < _targetAmount)
+		{
+			return false;
+		}
+
+		var privacyRankedPockets =
+			knownByRecipientPockets
+				.Select(pocket =>
+				{
+					var containedRecipientLabelsCount = pocket.Labels.Count(label => recipient.Contains(label, StringComparer.OrdinalIgnoreCase));
+					var totalPocketLabelsCount = pocket.Labels.Count();
+					var totalRecipientLabelsCount = recipient.Count();
+					var index = ((double)containedRecipientLabelsCount / totalPocketLabelsCount) + ((double)containedRecipientLabelsCount / totalRecipientLabelsCount);
+
+					return (acceptabilityIndex: index, pocket);
+				})
+				.OrderByDescending(tup => tup.acceptabilityIndex)
+				.ThenBy(tup => tup.pocket.Labels.Count())
+				.ThenByDescending(tup => tup.pocket.Amount)
+				.Select(tup => tup.pocket)
+				.ToArray();
+
+		var bestPockets = new List<Pocket>();
+		foreach (var p in privacyRankedPockets)
+		{
+			bestPockets.Add(p);
+
+			if (bestPockets.Sum(x => x.Amount) >= targetAmount)
+			{
+				break;
+			}
+		}
+
+		foreach (var p in bestPockets.OrderBy(x => x.Amount).ToImmutableArray())
```

Comment:
We should also order by number of labels here. See https://github.com/zkSNACKs/WalletWasabi/pull/7443#discussion_r823354315

```suggestion
		foreach (var p in bestPockets.OrderBy(x => x.Amount).ThenByDescending(x => x.Labels.Count()).ToImmutableArray())
```